Minor editorial, grammatical, and spelling improvments to docs (#1701)
vercel/turborepo: docs/pages/docs/faq.mdx

## Do I have to use Vercel to use Turborepo?

> No. Turborepo is an open source project and is not tied to any specific hosting provider or Remote Cache provider. The default Remote Cache provider is Vercel, should you opt-in to enable it. However, you can use any other provider that you like as long as they support the same API. There are several open source community Remote Caches that are compatible with Turborepo.

Reviewer: ```suggestion
No. Turborepo is an open-source project and is not tied to any specific hosting provider or Remote Cache provider. The default Remote Cache provider is Vercel, should you opt-in to enable it. However, you can use any other provider you like if they support the same API. Several open-source community Remote Caches are compatible with Turborepo.
```

Author: Interesting "open source" vs "open-source". I'm okay with either, but are we consistent about this everywhere?
